dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -133,20 +135,24 @@ void TypeLookupService::add_type_dependencies(const TypeIdentifier& type_id,
 bool TypeLookupService::type_object_in_cache(const TypeIdentifier& ti) const
 {
   ACE_GUARD_RETURN(ACE_Thread_Mutex, g, mutex_, false);
-  TypeMap::const_iterator it_type_id_with_size_seq = minimal_type_map_.find(ti);
-  return it_type_id_with_size_seq != minimal_type_map_.end();
+  return type_map_.find(ti) != type_map_.end();
 }
 
 bool TypeLookupService::extensibility(TypeFlag extensibility_mask, const TypeIdentifier& type_id) const
 {
   ACE_GUARD_RETURN(ACE_Thread_Mutex, g, mutex_, false);
   bool result = false;
-  TypeObject to = get_type_objects_i(type_id);
-
-  if (TK_UNION == to.minimal.kind) {
-    result = to.minimal.union_type.union_flags & extensibility_mask;
-  } else if (TK_STRUCTURE == to.minimal.kind) {
-    result = to.minimal.struct_type.struct_flags & extensibility_mask;
+  const TypeObject& to = get_type_objects_i(type_id);
+  TypeKind tk = to.kind == EK_MINIMAL ? to.minimal.kind : to.complete.kind;
```

Comment:
```suggestion
  const TypeKind tk = to.kind == EK_MINIMAL ? to.minimal.kind : to.complete.kind;
```

Reply:
`tk` is changed in a later for-loop so it can't be a constant.